File: scratch/run_audit.js
const fs = require('fs');
const path = require('path');

const excludeDirs = new Set(['.git', 'node_modules', 'node_modules_old', '.next', 'dist', 'build', '.gemini']);

let totalFiles = 0;
let totalSrcFiles = 0;
const extensionCounts = {};

function crawl(dir) {
  let files;
  try {
    files = fs.readdirSync(dir);
  } catch (e) {
    return; // Bypass files/folders that cannot be read
  }
  for (const file of files) {
    const fullPath = path.join(dir, file);
    let stat;
    try {
      stat = fs.statSync(fullPath);
    } catch (e) {
      continue; // Bypass stats error (corrupted reparse points)
    }
    if (stat.isDirectory()) {
      if (!excludeDirs.has(file)) {
        crawl(fullPath);
      }
    } else {
      totalFiles++;
      const ext = path.extname(file).toLowerCase();
      if (['.ts', '.tsx', '.js', '.jsx', '.css', '.html', '.prisma', '.json'].includes(ext)) {
        totalSrcFiles++;
      }
      extensionCounts[ext] = (extensionCounts[ext] || 0) + 1;
    }
  }
}

crawl('.');
console.log(`Total Files (excl. build/git/node_modules): ${totalFiles}`);
console.log(`Total Source Files: ${totalSrcFiles}`);
console.log("Extension distribution:");
Object.entries(extensionCounts)
  .sort((a, b) => b[1] - a[1])
  .forEach(([ext, count]) => {
    console.log(`  ${ext || '(no-ext)'}: ${count}`);
  });

File: scratch/run_audit.py
import os

def run_project_audit():
    total_files = 0
    total_src_files = 0
    extension_counts = {}
    
    # Exclude directories
    exclude_dirs = {'.git', 'node_modules', '.next', 'dist', 'build', '.gemini'}
    
    for root, dirs, files in os.walk('.'):
        # Filter directories in place
        dirs[:] = [d for d in dirs if d not in exclude_dirs]
        
        for file in files:
            total_files += 1
            ext = os.path.splitext(file)[1]
            if ext in {'.ts', '.tsx', '.js', '.jsx', '.css', '.html', '.prisma'}:
                total_src_files += 1
            extension_counts[ext] = extension_counts.get(ext, 0) + 1
            
    print(f"Total Files (excluding node_modules/next): {total_files}")
    print(f"Total Source Files: {total_src_files}")
    print("Extension distribution:")
    for ext, count in sorted(extension_counts.items(), key=lambda x: x[1], reverse=True):
        print(f"  {ext}: {count}")

run_project_audit()
